### PR TITLE
Block contrail dns service unless enabled

### PR DESCRIPTION
**Pull request by vpramo**
In v2.1 Contrail installs bind service on contrail controller this will disrupt the dnsmasq , hence adding a flag to disable contrail-dns as we are not using this service for now.

### CODE REVIEW

**Comment by vpramo** [comment]
@bodepd please check this. I have basically added a flag for disabling contrail-dns, and if enabled we can it listens on different port and have added a dnsmasq configuration for forwarding traffic based on domain name to contrail-dns.

**Comment by bodepd** [comment]
looks good to me. I assume that we are blocked on this until integration tests work again?

File: files/contrail-named.ini [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+autostart=true
+killasgroup=true
+stopsignal=KILL
+stdout_capture_maxbytes=1MB
+startsecs=5
+redirect_stderr=true
+stdout_logfile=/var/log/contrail/contrail-named-stdout.log
+stderr_logfile=/dev/null
+exitcodes=0                   ; 'expected' exit codes for process (default 0,2)
+user=contrail
```
Comment (sajuptpm) [inline files/contrail-named.ini]: repeated user=contrail

File: manifests/init.pp [hunk 2]
```diff
@@ -202,6 +202,14 @@
 #  Enable ifmap, defaults to true. Ifmap is not required to be setup in case of
 #  a separate node which is only running contrail analytics.
 #
+# [*enable_Dns*]
```
Comment (sajuptpm) [inline manifests/init.pp]: cap D

File: spec/classes/control_spec.rb [hunk 3]
```diff
@@ -29,16 +50,13 @@
...
-      should contain_file('/etc/contrail/contrail-control.conf').with_content(/user=10.1.1.1/)
-      should contain_service('contrail-control').with({
+      should contain_file('/etc/contrail/dns/named.conf').with_content(/listen-on port 10000/)
+      should contain_file('/etc/contrail/supervisord_control_files/contrail-named.ini')
+      should contain_service('contrail-named').with({
         'ensure'    => 'running',
         'enable'    => true,
-        'subscribe' => 'File[/etc/contrail/contrail-control.conf]',
-        'require'   => 'Package[contrail-control]'
+        'subscribe' => '[File[/etc/contrail/dns/named.conf]{:path=>"/etc/contrail/dns/named.conf"}, File[/etc/contrail/supervisord_control_files/contrail-named.ini]{:path=>"/etc/contrail/supervisord_control_files/contrail-named.ini"}]',
```
Comment (sajuptpm) [inline spec/classes/control_spec.rb]: break this log line

File: files/contrail-named.ini (hunk 1 above)
Comment (anshprat) [inline files/contrail-named.ini]: @vpramo need to fix the user dulicacy as saju pointed out.

Comment (anshprat) [inline files/contrail-named.ini]: +stderr_logfile=/dev/null 
Lets redirect it to an error file instead?

**Pull request merged**